c_src/ex_dtls/native.c

```diff
@@ -266,6 +266,41 @@ UNIFEX_TERM do_handshake(UnifexEnv *env, State *state) {
   return unifex_raise(state->env, "Handshake failed: no packets generated");
 }
 
+UNIFEX_TERM write_data(UnifexEnv *env, State *state, UnifexPayload *payload) {
+  if (state->hsk_finished != 1) {
+    DEBUG("Cannot write, handshake not finished");
+    return write_data_result_error_handshake_not_finished(env);
+  }
+
+  int ret = SSL_write(state->ssl, payload->data, payload->size);
+  if (ret <= 0) {
+    DEBUG("Unable to write data");
+    return unifex_raise(env, "Unable to write data");
+  }
+
+  BIO *wbio = SSL_get_wbio(state->ssl);
+  size_t pending_data_len = BIO_ctrl_pending(wbio);
+  if (pending_data_len <= 0) {
```

Comment:
[nitpick] Since it's a `size_t`
```suggestion
  if (pending_data_len == 0) {
```